relbench/external/nn.py

```diff
@@ -113,7 +113,13 @@ def forward(
         out_dict: Dict[NodeType, Tensor] = {}
 
         for node_type, time in time_dict.items():
-            rel_time = seed_time[batch_dict[node_type]] - time
+            try:
+                rel_time = seed_time[batch_dict[node_type]] - time
+            except:  # TODO (josh) this is a massive hack to fix an issue i came across with the
+                # F1 dataset and still dont understand
+                rel_time = (
+                    seed_time[batch_dict[node_type]] - time[batch_dict[node_type]]
+                )
```

Comment:
What's the error message if you don't have this? 

`try, except` without clear if-statement should be avoided.

Reply:
This is the message. Yes agreed this should be dealt with properly before being merged.

```
Traceback (most recent call last):
  File "examples/gnn.py", line 214, in <module>
    train_loss = train()
  File "examples/gnn.py", line 166, in train
    pred = model(
  File "/lfs/turing4/0/joshrob/miniconda3/envs/relbench/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/lfs/turing4/0/joshrob/miniconda3/envs/relbench/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "examples/gnn.py", line 141, in forward
    rel_time_dict = self.temporal_encoder(seed_time, time_dict, batch_dict)
  File "/lfs/turing4/0/joshrob/miniconda3/envs/relbench/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/lfs/turing4/0/joshrob/miniconda3/envs/relbench/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "/lfs/turing4/0/joshrob/graph/RDL/relbench/relbench/external/nn.py", line 117, in forward
    rel_time = seed_time[batch_dict[node_type]] - time
RuntimeError: The size of tensor a (22257) must match the size of tensor b (916) at non-singleton dimension 0
```